### PR TITLE
PUSH 23.0.3 INTO MAIN

### PR DESCRIPTION
**Pull request by ridz1208**
DO NOT SQUASH

See comments on each file for potential issues

CONFLICTS:
	both modified:   .github/workflows/loristest.yml
	both modified:   docker-compose.yml
	both modified:   jsx/Filter.js
	both modified:   jsx/FilterableDataTable.js
	both modified:   modules/battery_manager/test/TestPlan.md
	deleted by us:   modules/behavioural_qc/jsx/behavioural_qc_module.js
	both modified:   modules/dashboard/test/TestPlan.md
	both modified:   modules/datadict/php/datadict.class.inc
	both modified:   modules/datadict/test/TestPlan.md
	both modified:   modules/document_repository/jsx/categoryForm.js
	both modified:   modules/document_repository/php/uploadcategory.class.inc
	both modified:   modules/instrument_builder/jsx/react.questions.js
	both modified:   modules/instrument_manager/php/instrument_manager.class.inc
	both modified:   modules/survey_accounts/php/addsurvey.class.inc
	both modified:   php/libraries/NDB_BVL_Instrument.class.inc
	both modified:   php/libraries/NDB_BVL_Instrument_LINST.class.inc
	both modified:   src/Middleware/UserPageDecorationMiddleware.php
	both modified:   test/test_instrument/test_instrumentTest.php
	both modified:   test/unittests/NDB_BVL_Instrument_LINST_ToJSON_Test.php

### CODE REVIEW

File: .github/workflows/loristest.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
       uses: shivammathur/setup-php@v2
```
Comment (ridz1208) [inline .github/workflows/loristest.yml:24]: seems like php-ast extension might be getting installed twice ?

File: jsx/Filter.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, {Component} from 'react';
+import React, {useEffect} from 'react';
```
Comment (ridz1208) [inline jsx/Filter.js:1]: MAJOR collisions in this file mainly between the commits referenced below
main: 2372c4294cdfcdcce2843bbf6eb7a2d2df20a640
23: b52a8a9592041af2040fd9d121642619f53d9146

File: .eslintignore [hunk 3]
```diff
@@ -1,5 +1,7 @@
 node_modules/*
 vendor/*
+project/*
+p_*/*
```
Comment (driusan) [inline .eslintignore:4]: Where I remember project but where did `p_*/*` come from?

File: .github/workflows/loristest.yml (hunk 1 above)
Comment (driusan) [reply .github/workflows/loristest.yml:24]: yeah, looks like it. Remove the `sudo` part

File: jsx/FilterableDataTable.js [hunk 4]
```diff
@@ -24,40 +24,75 @@ class FilterableDataTable extends Component {
   constructor(props) {
```
Comment (ridz1208) [inline jsx/FilterableDataTable.js:24]: MAJOR collisions in this file mainly between the commits referenced below
main: 2372c4294cdfcdcce2843bbf6eb7a2d2df20a640, b9c25e8df2f029cdfb67c9698d4dbc290a060723
23: b52a8a9592041af2040fd9d121642619f53d9146

The reconciliation should be reviewed carefully as the bugfix and new feature both made major modifications

File: jsx/Filter.js (hunk 2 above)
Comment (driusan) [reply jsx/Filter.js:1]: @HenriRabalais looks like you're the author of both of the above commits, can you double check this file?

File: .eslintignore (hunk 3 above)
Comment (ridz1208) [reply .eslintignore:4]: that's particular to my setup, I use `p_projectname` to swap project directories. everything is returning errors its very annoying.

I'll remove that line

File: .github/workflows/loristest.yml (hunk 1 above)
Comment (ridz1208) [reply .github/workflows/loristest.yml:24]: do you know the fix for git actions ? is cant find php8-ast

File: .eslintignore (hunk 3 above)
Comment (driusan) [reply .eslintignore:4]: You can add personal ignores to `.git/info/exclude` for yourself.

File: .github/workflows/loristest.yml (hunk 1 above)
Comment (driusan) [reply .github/workflows/loristest.yml:24]: on the main branch it's just `php-ast`. Maybe it's easiest to just copy `.github/workflows/loristest.yml` from the main branch? Any fixes to it on the release branch were just to fix the release branch and it's currently passing on main..

**Comment by ridz1208** [comment]
@kongtiaowang could you please take a look at this as soon as possible?
let me know what might be causing the errors

File: php/libraries/NDB_BVL_Instrument.class.inc [hunk 5]
```diff
@@ -2764,7 +2850,33 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         if ($request->getMethod() === "POST" && !isset($req['ClearInstrument'])) {
             $this->save();
         }
-        return (new \Laminas\Diactoros\Response())
+
+        // Disable form if data entry is complete
+        $user      = $request->getAttribute("user");
+        $sessionID = $request->getQueryParams()["sessionID"];
+        $timepoint = \Timepoint::singleton($sessionID);
```
Comment (kongtiaowang) [inline php/libraries/NDB_BVL_Instrument.class.inc:2857]:
```suggestion
        $timepoint = \Timepoint::singleton(new SessionID($sessionID));
```

File: jsx/Filter.js [hunk 6]
```diff
@@ -113,73 +101,72 @@ class Filter extends Component {
...
-      </FormElement>
-    );
-  }
+        {filterActions}
+        {renderFilterFields()}
+        <ButtonElement
+          label="Clear Filters"
+          type="reset"
+          onUserInput={props.clearFilters}
+        />
```
Comment (HenriRabalais) [inline jsx/Filter.js:166]: This button should be removed since it's been implemented in the filterActions

Comment (HenriRabalais) [inline jsx/Filter.js:166]:
```suggestion
```

File: jsx/Filter.js [hunk 7]
```diff
@@ -113,73 +101,72 @@ class Filter extends Component {
...
-    );
+      );
+    };
+  };
+
+  const filterActions = (
+    <ul className='nav nav-tabs navbar-right' style={{borderBottom: 'none'}}>
+      {filterPresets()}
+      <li>
+        <a role='button' name='reset' onClick={props.clearFilter}>
```
Comment (HenriRabalais) [inline jsx/Filter.js:144]:
```suggestion
        <a role='button' name='reset' onClick={props.clearFilters}>
```

File: jsx/Filter.js [hunk 8]
```diff
@@ -113,73 +101,72 @@ class Filter extends Component {
...
-    const filterActions = (
-      <ul className='nav nav-tabs navbar-right' style={{borderBottom: 'none'}}>
-        {filterPresets()}
-        <li>
-          <a role='button' name='reset' onClick={this.props.clearFilter}>
-            Clear Filter
+  const filterPresets = () => {
+    if (props.filterPresets) {
+      const presets = props.filterPresets.map((preset) => {
+        const handleClick = () => props.updateFilter(preset.filter);
```
Comment (HenriRabalais) [inline jsx/Filter.js:120]:
```suggestion
        const handleClick = () => props.updateFilters(preset.filter);
```

File: jsx/FilterableDataTable.js [hunk 9]
```diff
@@ -104,19 +139,20 @@ class FilterableDataTable extends Component {
...
         id={this.props.name + '_filter'}
         columns={this.props.columns}
-        filter={filters}
-        fields={this.props.fields}
-        updateFilter={this.updateFilter}
-        clearFilter={this.clearFilter}
+        filters={filters}
         filterPresets={this.props.filterPresets}
+        fields={this.props.fields}
+        addFilter={this.addFilter}
```
Comment (HenriRabalais) [inline jsx/FilterableDataTable.js:145]:
```suggestion
        addFilter={this.addFilter}
        updateFilters={this.updateFilters}
```

**Comment by ridz1208** [comment]
@kongtiaowang still an error

**Comment by kongtiaowang** [comment]
@ridz1208 This will help you. 
Screen Shot 2021-04-07 at 11 10 48 AM

**Comment by ridz1208** [comment]
@kongtiaowang done
https://github.com/aces/Loris/pull/7418/files#diff-618cfcb420e1faf87fdd851ccdf0e1432c46e2536133b4cc8e1b4f9f5aa5569f

File: jsx/Filter.js (hunk 2 above)
Comment (HenriRabalais) [reply jsx/Filter.js:1]: @driusan done!

**Pull request merged**